### PR TITLE
Add ability to redeploy system-agent

### PR DESCRIPTION
**Pull request by StrongMonkey**
Context: https://github.com/rancher/rancher/issues/34107#issuecomment-899820635

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/34107

File: pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go
Comment (snasovich) [inline pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go:152]: Ideally we don't want this ENV at all if `redeploySystemAgentGeneration` is not set because it will result in a change of checksum and redeployment of system-agent if this change gets applied to cluster that was generated before the change. I know we never released this functionality, but it would be cleaner anyway to not add this env if it's not set on the Cluster CR. With current code I believe it would be set to 0.

File: pkg/apis/provisioning.cattle.io/v1/cluster.go
Comment (snasovich) [inline pkg/apis/provisioning.cattle.io/v1/cluster.go:32]: Alignment

Comment (StrongMonkey) [reply pkg/apis/provisioning.cattle.io/v1/cluster.go:32]: This actually passed go fmt.

File: pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go
Comment (StrongMonkey) [reply pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go:152]: yeah, I can change that.

File: pkg/apis/provisioning.cattle.io/v1/cluster.go
Comment (snasovich) [reply pkg/apis/provisioning.cattle.io/v1/cluster.go:32]: Yeah, didn't notice a newline - it's in own block so doesn't have to align.

File: pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go
Comment (StrongMonkey) [inline pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go:221]: After talking to @dweomer , we can use secretRef to force job rerun. This doesn't need SUC pr to support that anymore.

File: pkg/provisioningv2/kubeconfig/manager.go
Comment (StrongMonkey) [inline pkg/provisioningv2/kubeconfig/manager.go:225]: Checking the annotation seems a little bit problematic during backup and restore as the secret can be created just with no annotation first https://github.com/rancher/rancher/blob/e201d2e02ead6884d789024c7fbacf0819b68363/pkg/provisioningv2/kubeconfig/manager.go#L282. This causes secret to be created and deleted back and forth and somehow the token inside it become invalid. @nickgerace I feel like we can remove this as checking `token` field should suffice.

Comment (nickgerace) [reply pkg/provisioningv2/kubeconfig/manager.go:225]: Ahhh good call. That discrepancy makes sense. Let's do it.

**Review by nickgerace — APPROVED** [review]
LGTM with Sergey's convo

**Review by jiaqiluo — APPROVED** [review]
LGTM

File: pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go
Comment (snasovich) [inline pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go:236]: Previous code has `plan` as the first object in the result, this one would have it as last. This may not change anything functionally, but ideally we want to preserve the same order.

**Review by snasovich — CHANGES_REQUESTED** [review]

File: pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go
Comment (StrongMonkey) [reply pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go]: :+1:

**Review by snasovich — APPROVED** [review]
LGTM, please get 1 committer approval (@dramich or @kinarashah).

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
